pkg/ipamd/ipamd.go

```diff
@@ -988,8 +997,24 @@ func (c *IPAMContext) nodeIPPoolReconcile(interval time.Duration) {
 
 func (c *IPAMContext) eniIPPoolReconcile(ipPool []string, attachedENI awsutils.ENIMetadata, eni string) {
```

Comment:
This method looks super big. Might need some refactoring to divide this into smaller chunks. thoughts?

Reply:
Split out the middle for-loop